src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -5,7 +5,7 @@
 /**
  * Parameterize this service and add its implementation in a separate class.
  */
-public interface MachineService {
+public interface MachineService<T> {
```

Comment:
```suggestion
public interface MachineService<T extends Machine> {
```

Reply:
This comment from Oleh is important, otherwise user may try to create MachineService parametrized with Animal or Plant etc.